radio/src/pulses/afhds3.cpp

```diff
@@ -0,0 +1,790 @@
+/*
+ * Copyright (C) OpenTX
+ *
+ * Based on code named
+ *   th9x - http://code.google.com/p/th9x
+ *   er9x - http://code.google.com/p/er9x
+ *   gruvin9x - http://code.google.com/p/gruvin9x
+ *
+ * License GPLv2: http://www.gnu.org/licenses/gpl-2.0.html
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "afhds3.h"
+#include "../debug.h"
+#include "../definitions.h"
+#include <cstdio>
+#define FAILSAFE_HOLD 1
+#define FAILSAFE_CUSTOM 2
+#define MAX_RETRIES_AFHDS3 5
+
+extern int32_t getChannelValue(uint8_t channel);
```

Comment:
would be better in a .h if you introduce a function for it
I think it could be even inlined

Reply:
Declaration of this function can be done in afhds3.h however it belongs to global namespace and is defined in pulses.cpp - doing it in pulses.h will case circular referencing.
For now I just declare it in afhds3.h.